src/GafferCycles/IECoreCyclesPreview/ShaderNetworkAlgo.cpp

```diff
@@ -773,3 +775,354 @@ IECoreScene::ShaderNetworkPtr convertLightShader( const IECoreScene::ShaderNetwo
+const InternedString g_vector1Parameter( "vector1" );
+const InternedString g_vector2Parameter( "vector2" );
+const InternedString g_widthParameter( "width" );
+
+void transferUSDLightParameters( ShaderNetwork *network, InternedString shaderHandle, const Shader *usdShader, Shader *shader )
+{
+	Color3f color = parameterValue( usdShader, g_colorParameter, Color3f( 1 ) );
+	if( parameterValue( usdShader, g_enableColorTemperatureParameter, false ) )
+	{
+		color *= blackbody( parameterValue( usdShader, g_colorTemperatureParameter, 6500.0f ) );
+	}
+	shader->parameters()[g_colorParameter] = new Color3fData( color );
+
+	transferUSDParameter( network, shaderHandle, usdShader, g_exposureParameter, shader, g_exposureParameter, 0.0f );
+	transferUSDParameter( network, shaderHandle, usdShader, g_intensityParameter, shader, g_intensityParameter, 1.0f );
+	transferUSDParameter( network, shaderHandle, usdShader, g_normalizeParameter, shader, g_normalizeParameter, false );
+	transferUSDParameter( network, shaderHandle, usdShader, g_shadowEnableParameter, shader, g_castShadowParameter, true );
+}
+
+void transferUSDShapingParameters( ShaderNetwork *network, InternedString shaderHandle, const Shader *usdShader, Shader *shader )
+{
+	if( auto d = usdShader->parametersData()->member<FloatData>( g_shapingConeAngleParameter ) )
+	{
+		shader->setName( "spot_light" );
+		shader->parameters()[g_spotAngleParameter] = new FloatData( d->readable() * 2.0f );
+		/// \todo This logic is copied from `IECoreArnold::ShaderNetworkAlgo` for reasons
+		/// documented there, but I suspect things have moved on a bit and we can probably
+		/// take this at face value now.
+		if( parameterValue( usdShader, g_shapingConeSoftnessParameter, 0.0f ) > 1.0 )
+		{
+			IECore::msg( IECore::Msg::Warning, "transferUSDShapingParameters", "Ignoring `shaping:cone:softness` as it is greater than 1" );
+		}
+		else
+		{
+			transferUSDParameter( network, shaderHandle, usdShader, g_shapingConeSoftnessParameter, shader, g_spotSmoothParameter, 0.0f );
+		}
+	}
+}
+
+// Should be called after `transferUSDLightParameters()`, as it needs to examine
+// the transferred `color` parameter.
+void transferUSDTextureFile( ShaderNetwork *network, InternedString shaderHandle, const Shader *usdShader, const Shader *shader )
+{
+	const string textureFile = parameterValue( usdShader, g_textureFileParameter, string() );
+	if( textureFile.empty() )
+	{
+		return;
+	}
+
+	ShaderPtr imageShader;
+	if( usdShader->getName() == "DomeLight" )
+	{
+		imageShader = new Shader( "environment_texture", "cycles:shader" );
+		string format = parameterValue( usdShader, g_textureFormatParameter, string( "equirectangular" ) );
+		if( format == "mirroredBall" )
+		{
+			format = "mirror_ball";
+		}
+		else if( format == "latlong" )
+		{
+			format = "equirectangular";
+		}
+		else
+		{
+			IECore::msg( IECore::Msg::Warning, "convertUSDShaders", fmt::format( "Unsupported value \"{}\" for DomeLight.format", format ) );
```

Comment:
Done in https://github.com/GafferHQ/gaffer/pull/5400/commits/140d3b96367a67fe4f37afa2aeb589c53bd37df4. I also defaulted `cone:angle` to something other than 90, so that when you first turn it on you're not confronted with a weird disk-looking thing.